server/src/utils.ts

```diff
@@ -113,12 +113,14 @@ export function lspRangeToTsPositions(
  * @param relatedInfo
  */
 export function tsRelatedInformationToLspRelatedInformation(
-    scriptInfo: ts.server.ScriptInfo,
+    projectService: ts.server.ProjectService,
     relatedInfo?: ts.DiagnosticRelatedInformation[]): lsp.DiagnosticRelatedInformation[]|undefined {
   if (relatedInfo === undefined) return;
   const lspRelatedInfo: lsp.DiagnosticRelatedInformation[] = [];
   for (const info of relatedInfo) {
     if (info.file === undefined || info.start === undefined || info.length === undefined) continue;
+    const scriptInfo = projectService.getScriptInfo(info.file.fileName);
+    if (scriptInfo === undefined) continue;
```

Comment:
There is an issue here. The diagnostic file may differ from the one in 'info', so the range needs to be adjusted using the diagnostic file.

Reply:
I’m not quite following. The diagnostic isn’t already mapped to the appropriate info by this time? Should it be done in the language service?

Reply:
This issue is the transformation from the ts range to lsp range.

For example:

The diagnostic is located in the HTML, and there is a related diagnostic for the `ts`(Error occurs in the template of component ***). For the old logic, the script info is the HTML, but we should use the ts script info to transform the ts range. 

And we should not assume here that the related diagnostics all come from the same script info.

Reply:
Right, the related info can be a different file. I’m not sure then that I follow this comment thread. I don’t see those edits here. Are you unsure if they’re right or saying there’s no way to fix it?

Reply:
Got it! I understand now. Thanks for this.